stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/LongClickFriendlyLinkMovementMethod.kt

```diff
@@ -0,0 +1,53 @@
+package io.getstream.chat.android.ui.utils
+
+import android.text.method.LinkMovementMethod
+import android.view.View
+import android.widget.TextView
+import androidx.core.widget.doAfterTextChanged
+import com.getstream.sdk.chat.utils.Utils
+import io.getstream.chat.android.ui.utils.LongClickFriendlyLinkMovementMethod.Companion.set
+
+/**
+ * A customized [LinkMovementMethod] implementation that lets you handle links being
+ * clicked with [onLinkClicked], while also keeping long clicks as a separate event,
+ * forwarded to [longClickTarget].
+ *
+ * This class is reattached to the target [textView] every time its content changes,
+ * so you only need to call [set] one time.
+ */
+internal class LongClickFriendlyLinkMovementMethod private constructor(
+    private val textView: TextView,
+    private val longClickTarget: View,
+    private val onLinkClicked: (url: String) -> Unit,
+) : Utils.TextViewLinkHandler() {
```

Comment:
First time I see such a framework feature. Looks interesting 👍

Reply:
Yeah, the default `LinkMovementMethod` doesn't handle long clicks correctly so, together with Marton we had to workaround that somehow